lib/components/key-handler.js

```diff
@@ -149,20 +150,38 @@ export function keyToggleHandler({keyCode, keyName, keyEventName}: DecoratorProp
       }
 
       renderDecoratedComponent(): ReactElement {
-        const {keyCode, keyName} = this.state;
+        const {keyValue, keyCode, keyName} = this.state;
 
-        return React.createElement(component, { ...this.props, keyCode, keyName });
+        return React.createElement(component, { ...this.props, keyValue, keyCode, keyName });
       }
 
       handleToggleKey(event: KeyboardEvent): void {
-        if (this.state.keyCode === event.keyCode) {
-          return this.setState({ keyCode: null, keyName: null });
+        if (!checkKeyValues(event, this.state)) {
+          return this.setState({ keyValue: null, keyCode: null, keyName: null });
         }
 
-        this.setState({ keyCode: event.keyCode, keyName: keyNames(event.keyCode) });
+        this.setState({ keyValue: event.key || keyNameVals(keyNames(event.keyCode)), keyCode: event.keyCode, keyName: keyNames(event.keyCode) });
       }
     };
   };
 }
 
+/**
+ * Helpers
+ */
+function valFromCode(keyCode: ?number): ?string {
+  if (!keyCode) {
+    return;
```

Comment:
Is `keyCode` ever a falsey value? 
* If yes should this return `'Unidentified'`? 
* If no can't we remove this `if` block and always return a `string`?